src/lib/plugins/login.js

```diff
@@ -234,11 +226,15 @@ module.exports.player = async function (player, serv, settings) {
     player.updateAndSpawn()
 
     announceJoin()
+    // mineflayer emits spawn event on health update so it needs to be done as last step
+    player.updateHealth(player.health)
     player.emit('spawned')
 
     await player.waitPlayerLogin()
     player.sendRestMap()
     sendChunkWhenMove()
+
+    player.save()
```

Comment:
by doing this a new entry in playerdata/ appears immediately after a new player joins